scroll-search.lic

```diff
@@ -0,0 +1,459 @@
+=begin
+  Documentation: TBD
+=end
+
+custom_require.call(%w(common common-items drinfomon))
+
+# Define Scroll Info class
+class ScrollInfo
+  def initialize(container, index, container_index, ordinal, spell, type)
+    # Instance variables
+    @container = container
+    @index = index
+	@container_index = container_index
+    @ordinal = ordinal
+    @spell = spell
+    @type = type
+  end
+
+  def get_container
+    return @container
+  end
+
+  def get_spell
+    return @spell
+  end
+
+  def get_ordinal
+    return @ordinal
+  end
+  
+  def get_index
+	return @index
+  end
+  
+  # represents the local index of the
+  # scroll within the container
+  def get_container_index
+	return @container_index
+  end
+
+  def get_type
+    return @type
+  end
+
+  def set_container(container)
+    @container = container
+  end
+  
+  def set_container_index(index)
+	@container_index = index
+  end
+  
+  def set_ordinal(ordinal)
+	@ordinal = ordinal
+  end
+end
+
+class ScrollSearch
+  include DRC
+  include DRCI
+
+  def initialize
+
+    # Variables
+    $scroll_count = 0
+	$global_scroll_index = if UserVars.scroll_counter.nil? then 0 else UserVars.scroll_counter end
+    $owned_scrolls = if UserVars.owned_scrolls.empty? then {} else UserVars.owned_scrolls end
+    
+	# RESET ALL
+	# UserVars.owned_scrolls = { }
+	# $owned_scrolls = { }
+	# UserVars.scroll_counter = 0
+
+    $settings = get_settings()
+    $debug_mode = $settings.debug_mode
+
+    echo("Settings debug mode is: #{$debug_mode}") if $debug_mode
+    if $settings.nil? || $settings.empty?
+      echo("Settings was nil!") if $debug_mode
+    end
+
+    arg_definitions = 
+    [
+      [
+        { name: 'action', regex: /(^dis.*|^sea.*)/i, optional: false, description: 'Whether to SEARCH | DISPLAY your container for spell scrolls. ' },
+        { name: 'container', regex: /\w+/i, optional: true, description: 'Searches given container and labels any unlabeled scrolls. ' }
+      ]
+    ]
+
+    args = parse_args(arg_definitions)
+
+    if args.to_h.empty?
+      display_args(arg_definitions)
+      return
+    end
+
+    pause
+
+    case args.action
+    when /^sea.*/i

```

Comment:
Rather than doing regex matching, you might want to consider two optional args (search and display, similar to how you did container), then you could act on `args.search` or `args.display`. The `if args.to_h.empty?` block above could then be `unless args.search || args.display` and display your error message.